### DISCUSSION TITLE
Rename implicit many to many generated table columns

### DISCUSSION THREAD

**Discussion by varshith5c** [prisma/prisma#18501] [Q&A]
## Problem

How to change the column name(A and B) of implicit many to many tables?

## Suggested solution

It would be great if we can specify column names in the prisma schema or give the primary key column between two tables as the column name. A and B causes lots of issues.

**Comment by janpio** [comment]
What issues does this cause?

**Reply by varshith5c** [reply]
These column names makes it difficult to read when working in multiple teams like product and data team. If the column names were much more verbose then it would be easier for the ones who only work in data team.

**Reply by janpio** [reply]
Problem here is that there is no place to configure implicit many to many relations. They are only ... implicitly created, and there is nothing explicit about them in the Prisma schema where you could put that information.

Do you have a suggestion where we could put such explicit information?

**Reply by varshith5c** [reply]
We can use `prisma migrate dev --create-only ` and then edit the auto generated column names . I dont know whether it has any other bad implications or not.

From this :
```
CREATE TABLE "_post" (
    "A" TEXT NOT NULL,
    "B" TEXT NOT NULL
);


CREATE TABLE "_category" (
    "A" TEXT NOT NULL,
    "B" TEXT NOT NULL
); 
```

 To this:
```
CREATE TABLE "_post" (
    "postId" TEXT NOT NULL,
    "categoryId" TEXT NOT NULL
);


CREATE TABLE "_category" (
    "categoryId" TEXT NOT NULL,
    "postId" TEXT NOT NULL
); 
```

As an alternate option , something like the following prisma schema would be helpful:

```
model post {
  id     String   @id @default(uuid())
  name   String
  category category[] @relation("TaskToPeople")

  @@map([["A", "postId"], ["B", "categoryId"]])
}

model category {
  id   String @id @default(uuid())
  name String
  post post[] @relation("TaskToPeople")

  @@map([["A", "categoryId"], ["B", "postId"]])
  
}
```

**Reply by nurul3101** [reply]
Thanks for sharing the suggestions.
For reference here is a Feature Request for the same, please feel free to add your suggestions there as well.

- https://github.com/prisma/prisma/issues/10243
